### PR TITLE
feat(Table): expand checkbox click area

### PR DESCRIPTION
**Pull request by Bluepuper**
Issue: [#648](https://github.com/gravity-ui/uikit/issues/648)

### CODE REVIEW

**Comment by teleginzhenya** [comment]
@krozhkov can you take a look on this pr?

**Comment by teleginzhenya** [comment]
@amje maybe you? lgtm :)

File: src/components/Table/Table.scss
Comment (amje) [inline src/components/Table/Table.scss:63]: checkbox width is 17px

File: src/components/Table/hoc/withTableSelection/withTableSelection.tsx
Comment (amje) [inline src/components/Table/hoc/withTableSelection/withTableSelection.tsx:93]: Not used anywhere

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Table/Table.scss
Comment (Bluepuper) [reply src/components/Table/Table.scss]: done

File: src/components/Table/hoc/withTableSelection/withTableSelection.tsx
Comment (Bluepuper) [reply src/components/Table/hoc/withTableSelection/withTableSelection.tsx]: removed

**Comment by Bluepuper** [comment]
@amje Can you check please?

**Comment by amje** [comment]
@Bluepuper 
There are some issues when Table is overflowing

**Comment by Bluepuper** [comment]
> @Bluepuper There are some issues when Table is overflowing 

Add min-width of checkbox to cell, so it will not be shrinking

File: src/components/Table/Table.scss
Comment (amje) [inline src/components/Table/Table.scss:55]: Let's put these styles to the separate class. Class can be passed in enhanceColumns method

File: src/components/Table/hoc/withTableSelection/withTableSelection.scss
Comment (amje) [inline src/components/Table/hoc/withTableSelection/withTableSelection.scss:7]: What about styles when `edgePadding: false`?

File: src/components/Table/hoc/withTableSelection/withTableSelection.scss
Comment (amje) [inline src/components/Table/hoc/withTableSelection/withTableSelection.scss:2]: That will duplicate the Table styles

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Table/Table.scss
Comment (Bluepuper) [reply src/components/Table/Table.scss]: [separete](https://github.com/gravity-ui/uikit/pull/654/commits/7912e104afbadd045a107d072087d0f60272a472#diff-934636fcc01112fd9d8f251ed6903afc4d5a1b202f44f1828abd5b1f4e07e925R4) class

[use](https://github.com/gravity-ui/uikit/pull/654/commits/7912e104afbadd045a107d072087d0f60272a472#diff-e550fc4a251be96df10c117d97a0b9525d3ed80d4177be7d2169fa7518e4fa0cR165) in `enhanceColumns` method

File: src/components/Table/hoc/withTableSelection/withTableSelection.scss
Comment (Bluepuper) [reply src/components/Table/hoc/withTableSelection/withTableSelection.scss]: [remove](https://github.com/gravity-ui/uikit/pull/654/commits/7912e104afbadd045a107d072087d0f60272a472#diff-934636fcc01112fd9d8f251ed6903afc4d5a1b202f44f1828abd5b1f4e07e925L6) duplicating styles

File: src/components/Table/hoc/withTableSelection/withTableSelection.scss
Comment (Bluepuper) [reply src/components/Table/hoc/withTableSelection/withTableSelection.scss]: remove

File: src/components/Table/Table.tsx
Comment (Bluepuper) [inline src/components/Table/Table.tsx:155]: did this to remove double wrapping of checkbox so `padding: inherit` will work

**Comment by Bluepuper** [comment]
Can you check it please ?) @amje

File: src/components/Table/Table.scss
Comment (amje) [inline src/components/Table/Table.scss:53]: We can now move these styles to the `.checkbox_cell` class

File: src/components/Table/Table.tsx
Comment (amje) [reply src/components/Table/Table.tsx:155]: But now selection hoc "data" leaked to the base Table. Let's find a workaround here

File: src/components/Table/Table.scss
Comment (Bluepuper) [reply src/components/Table/Table.scss]: [moved](https://github.com/gravity-ui/uikit/pull/654/commits/2861dfe4d12e6258b3abfa5d6d9a233a14e1dc5a)

File: src/components/Table/Table.tsx
Comment (Bluepuper) [reply src/components/Table/Table.tsx:155]: this is one with additional wrapper
Screenshot 2023-10-10 at 21 25 18

this is one without
Screenshot 2023-10-10 at 21 26 24

Comment (Bluepuper) [reply src/components/Table/Table.tsx:155]: If I add `padding: inherit` to the `th-content` class, I can fix the existence of the extra wrapper, but then all the other cells are broken
Screenshot 2023-10-10 at 21 27 38

Comment (Bluepuper) [reply src/components/Table/Table.tsx:155]: i can add different wrapper for cell with `id === '_selection'`, something like this:
`if (id === '_selection') return <span  className={b('th-content_selection')}>content</span>;`

Comment (amje) [reply src/components/Table/Table.tsx:155]: I think we can remove this wrapper (`th-content`) completely, and add these styles instead:
```scss
    &__head &__cell {
        @include mixins.text-accent;
        &::first-letter {
            text-transform: uppercase;
        }
    }
```

Comment (Bluepuper) [reply src/components/Table/Table.tsx]: > I think we can remove this wrapper (`th-content`) completely, and add these styles instead:
> 
> ```css-scss
>     &__head &__cell {
>         @include mixins.text-accent;
>         &::first-letter {
>             text-transform: uppercase;
>         }
>     }
> ```

like [this](https://github.com/gravity-ui/uikit/pull/654/commits/af2ecc18111eb3e5907729e27dd091117f70a923) ?

Comment (Bluepuper) [reply src/components/Table/Table.tsx]: @amje

**Review by amje — APPROVED** [review]

**Comment by amje** [comment]
> Is it ok that checkbox isn't in center of a cell anymore? Screen Shot 2023-11-21 at 11 04 27

I assume, yes. Now it's an equal padding between all cells

**Comment by EgorKluch** [comment]
> @EgorKluch I assume, yes. Now it's an equal padding between all cells

But then needs to center text [too](https://st.yandex-team.ru/DATAUI-1598#655c913130316b33e3aac159).

I'm waiting for the final decision.

**Comment by amje** [comment]
@EgorKluch Oh, I thought you're talking about horizontal alignment. Checkbox should be centered as well when `verticalAlign: center` is set

**Comment by Bluepuper** [comment]
@amje `vertical-align: middle` doesn't work somewhy, [make PR with fix](https://github.com/gravity-ui/uikit/pull/1149) using `display: flex` is that ok ?

**Comment by EgorKluch** [comment]
Also change column height in some cases. I send links of example in telegram.
image

**Comment by Bluepuper** [comment]
> Also change column height in some cases. I send links of example in telegram. image

It turns out that the row height was wrong before this pr based on the [design](https://gravity-ui.com/design/guides/table). This pr also fixes that, now it is correct: 40px height + 1px bottom border. Before it was 40.5 px inner height + 1px border bottom

**Pull request merged**
